crates/re_viewer_context/src/gpu_bridge/colormap.rs

```diff
@@ -98,30 +103,53 @@ fn colormap_variant_ui(
 pub fn colormap_dropdown_button_ui(
     render_ctx: Option<&re_renderer::RenderContext>,
     ui: &mut egui::Ui,
-    map: &mut re_types::components::Colormap,
-) -> egui::InnerResponse<Option<egui::Response>> {
-    let selected_text = map.to_string();
-    let content_ui = |ui: &mut egui::Ui| {
-        let mut iter = re_types::components::Colormap::variants().iter();
+    map: &mut MaybeMutRef<'_, re_types::components::Colormap>,
+) -> egui::Response {
+    if let Some(map) = map.as_mut() {
+        let selected_text = map.to_string();
+        let content_ui = |ui: &mut egui::Ui| {
+            let mut iter = re_types::components::Colormap::variants().iter();
 
-        let Some(first) = iter.next() else {
-            return ui.label("<no variants>");
-        };
+            let Some(first) = iter.next() else {
+                return ui.label("<no variants>");
+            };
 
-        let mut response = colormap_variant_ui(render_ctx, ui, first, map);
+            let mut response = colormap_variant_ui(render_ctx, ui, first, map);
 
-        for option in iter {
-            response |= colormap_variant_ui(render_ctx, ui, option, map);
-        }
+            for option in iter {
+                response |= colormap_variant_ui(render_ctx, ui, option, map);
+            }
 
-        response
-    };
+            response
+        };
 
-    egui::ComboBox::from_id_source("color map select")
-        .selected_text(selected_text)
-        .show_ui(ui, |ui| {
-            list_item::list_item_scope(ui, "inner_scope", content_ui)
-        })
+        let mut inner_response = egui::ComboBox::from_id_source("color map select")
+            .selected_text(selected_text)
+            .show_ui(ui, |ui| {
+                list_item::list_item_scope(ui, "inner_scope", content_ui)
+            });
+        if let Some(inner) = inner_response.inner {
+            if inner.changed() {
+                inner_response.response.mark_changed();
+            }
+        }
+        inner_response.response
+    } else {
+        let map: re_types::components::Colormap = **map;
+        if let Some(render_ctx) = render_ctx {
+            ui.list_item_flat_noninteractive(
```

Comment:
does that make this a full width list item instead of just the content/right side of a list item? If so the function name or at least doc should call this out since it means we no longer can use this in a value_fn context.
(related to lib.rs comment on multiline editor issue)